bindings/matrix-sdk-ffi/src/room.rs

```diff
@@ -505,13 +505,30 @@ impl Room {
     pub async fn typing_notice(&self, is_typing: bool) -> Result<(), ClientError> {
         Ok(self.inner.typing_notice(is_typing).await?)
     }
+    pub async fn subscribe_to_typing_notifications(
+        self: Arc<Self>,
+        listener: Box<dyn TypingNotificationsListener>,
+    ) -> Arc<TaskHandle> {
+        Arc::new(TaskHandle::new(RUNTIME.spawn(async move {
+            let (_guard, mut subscriber) = self.inner.subscribe_to_typing_notifications();
+            while let Ok(typing_users) = subscriber.recv().await {
+                let typing_users = typing_users.iter().map(|u| u.to_string()).collect();
+                listener.call(typing_users);
+            }
```

Comment:
1. `_guard`: which guard? A longer name would be better for the comprehension,
2. `typing_users.iter()` produces an iterator over the slice, where the iterator's items are reference (see [`std::slice::Iter`](https://doc.rust-lang.org/std/slice/struct.Iter.html#impl-Iterator-for-Iter%3C'a,+T%3E). Since you're sure `typing_users` is going to be consumed entirely, you can use [`IntoIterator::into_iter`](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.into_iter-2) instead,
3. Finally, please use long variable names: `u` must be `user_id`.

```suggestion
            let (_event_handler_drop_guard, mut subscriber) = self.inner.subscribe_to_typing_notifications();

            while let Ok(typing_users) = subscriber.recv().await {
                let typing_users = typing_users.into_iter().map(|user_id| user_id.to_string()).collect();

                listener.call(typing_users);
            }
```